### PR TITLE
Fix the failing (internal) Key Vault cert-related tests.

### PR DESCRIPTION
**Pull request by adarce**
## Description
The `Get-X509FromSecret` helper function was making the wrong assumption that the leaf certificate in a chain would be located at the beginning of the secret returned by the Key Vault service. Some cert-related tests started to fail when the Key Vault service changed the ordering so that the leaf certificate is now located at the end of the secret. This was not a breaking change, as the Key Vault service does not guarantee any ordering of the certs in the chain.

To address this issue, this change instead searches the entire collection of certs in the secret by subject distinguished name (we replace the `Get-X509FromSecret` helper with `Get-X509FromSecretBySubjectDistinguishedName`). This increases test resiliency by not relying on any cert ordering.

This change also extends the `NotAfter` property of a hardcoded test cert that 3 tests rely on by 1000 years, since there were test failures due to the cert having expired in October 2016.

This fixes issue #3288 

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md).**
- [x] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/dev/documentation/platyPSHelp-documentation.md).**
- [x] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/dev/documentation/breaking-changes.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [x] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://gist.github.com/markcowl/338e16fe5c8bbf195aff9f8af0db585d#what-is-the-change).
- [x] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThrough` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [x] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [x] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [x] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@adarce__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (adarce). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by cormacpayne** [comment]
LGTM once KeyVault team signs off

File: src/ResourceManager/KeyVault/Commands.KeyVault.Test/Scripts/VaultCertificateTests.ps1 [hunk 1]
```diff
@@ -5,14 +5,15 @@ function Get-CertificateSubjectName([string]$certificateName)
...
+function Get-X509FromSecretBySubjectName([string]$keyVault, [string]$secretName, [string]$subjectName)
 {
     $secret = Get-AzureKeyVaultSecret $keyVault $secretName
     $secretValueBytes = [System.Convert]::FromBase64String($secret.SecretValueText)
     $x509FromSecretCollection = New-Object System.Security.Cryptography.X509Certificates.X509Certificate2Collection
     $x509FromSecretCollection.Import($secretValueBytes)
-    $x509FromSecret = $x509FromSecretCollection[0]
-    return $x509FromSecret
+    $x509Matches = $x509FromSecretCollection.Find([Security.Cryptography.X509Certificates.X509FindType]::FindBySubjectName, $subjectName, $false)
+    Assert-True { $x509Matches.Count -eq 1 }  # There should be exactly one cert in the collection with the given subject name.
```
Comment (rohmano) [inline src/ResourceManager/KeyVault/Commands.KeyVault.Test/Scripts/VaultCertificateTests.ps1]: Can we return null here if certificate not found? Let the test assert according to context.

**Comment by adarce** [comment]
@rohmano Thanks, I've updated my PR to address your feedback.

**Comment by rohmano** [comment]
Approved, GTM.

**Comment by shahabhijeet** [comment]
@adarce can you point me if the test that you fixed is part of CI? Searching "X509FromSecretBySubjectDistinguishedName"  or "Get-X509FromSecretBySubjectDistinguishedName" did not result in any match?
Is this test enabled?

**Comment by adarce** [comment]
@shahabhijeet The tests that I fixed are not part of CI, as they are only used internally by the Key Vault team.

**Comment by cormacpayne** [comment]
@adarce are we waiting on any other sign-offs?

**Comment by adarce** [comment]
@cormacpayne No, please feel free to merge.

**Pull request merged**